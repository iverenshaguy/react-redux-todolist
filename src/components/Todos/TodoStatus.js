import React from 'react';
import PropTypes from 'prop-types';
import Button from '../shared/Button';

const TodoStatus = props => {
  let buttonText;
  if (props.todo.completed) {
    buttonText = 'Mark as Undone';
  } else {
    buttonText = 'Mark as Done';
  }

  return (
    <Button
      buttonType="button"
      btnClass="btn btn-outline-primary"
      buttonText={buttonText}
      onClick={props.onClick}
    />
  );
};

TodoStatus.proptypes = {
  todo: PropTypes.object.isRequired,
  onEdit: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired
};

export default TodoStatus;
