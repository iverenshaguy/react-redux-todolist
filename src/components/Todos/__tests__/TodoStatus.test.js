import TodoStatus from '../TodoStatus';
import { todos as mockTodos } from '../../../__mocks__/todos.mock';

describe('TodoStatus', () => {
  const clickFunction = jest.fn();

  it('renders correctly', () => {
    const wrapper = shallow(
      <TodoStatus todo={mockTodos[1]} onClick={clickFunction} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('mounts correctly', () => {
    const wrapper = mount(
      <TodoStatus todo={mockTodos[1]} onClick={clickFunction} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('calls a function on click', () => {
    const wrapper = shallow(
      <TodoStatus todo={mockTodos[3]} onClick={clickFunction} />
    );

    wrapper.simulate('click');
    expect(clickFunction).toHaveBeenCalled();
  });

  it("shows 'Mark as Undone' according to passed in prop", () => {
    const wrapper = shallow(
      <TodoStatus todo={mockTodos[3]} onClick={clickFunction} />
    );
    expect(wrapper.html()).toContain('Mark as Undone');
  });

  it("shows 'Mark as Done' according to passed in prop", () => {
    const wrapper = shallow(
      <TodoStatus todo={mockTodos[2]} onClick={clickFunction} />
    );
    expect(wrapper.html()).toContain('Mark as Done');
  });
});
