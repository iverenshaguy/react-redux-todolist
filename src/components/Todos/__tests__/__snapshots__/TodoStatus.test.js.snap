// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoStatus mounts correctly 1`] = `
<TodoStatus
  onClick={[Function]}
  todo={
    Object {
      "completed": false,
      "id": 2,
      "title": " Who will be easy for those services ",
    }
  }
>
  <Button
    btnClass="btn btn-outline-primary"
    buttonText="Mark as Done"
    buttonType="button"
    onClick={[Function]}
  >
    <button
      className="btn btn-outline-primary"
      onClick={[Function]}
      type="button"
    >
      Mark as Done
    </button>
  </Button>
</TodoStatus>
`;

exports[`TodoStatus renders correctly 1`] = `
<Button
  btnClass="btn btn-outline-primary"
  buttonText="Mark as Done"
  buttonType="button"
  onClick={[Function]}
/>
`;
